src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+                    }
+                    var storageCredentialsFactory = CreateStorageCredentialsFactory(destinationUri.Uri);
+                    var parameters = new UploadParameters(destinationUri, null, filePath, true, NumberOfUploaderThreads <= 0 ? DefaultNumberOfUploaderThreads : NumberOfUploaderThreads)
+                    {
+                        Cmdlet = this,
+                        BlobObjectFactory = new CloudPageBlobObjectFactory(storageCredentialsFactory, TimeSpan.FromMinutes(1))
+                    };
+                    var vhdUploadContext = VhdUploaderModel.Upload(parameters);
+                    WriteObject(vhdUploadContext);
+                    uploadedDiskBlobs.Add(vhdUploadContext.DestinationUri);
+                }
+                this.DiskLink = uploadedDiskBlobs.Select(a => a.ToString()).ToArray();
+            }
+            else
+            {
+                WriteVerbose("DiskFile input is empty or null; skip disk file uploading...");
+            }
+        }
+        
+        private StorageCredentialsFactory CreateStorageCredentialsFactory(Uri pageBlobUrl)
+        {
+            if (pageBlobUrl == null)
+            {
+                throw new ArgumentNullException("pageBlobUrl");
+            }
+            StorageCredentialsFactory storageCredentialsFactory;
+            if (pageBlobUrl != null && StorageCredentialsFactory.IsChannelRequired(pageBlobUrl))
+            {
+                storageCredentialsFactory = new StorageCredentialsFactory(this.ResourceGroupName, this.storageClient, DefaultContext.Subscription);
+            }
+            else
+            {
+                storageCredentialsFactory = new StorageCredentialsFactory();
+            }
+
+            return storageCredentialsFactory;
+        }
+
+        private void ThrowIfOSOrDataDisksNotExist()
+        {
+            if (NoDiskLinkExistenceCheck)
+            {
+                WriteVerbose("Skip checking disk link's existence...");
+                return;
+            }
+
+            foreach (var blobUrl in this.DiskLink)
+            {
+                if (!CheckIfPageBlobExists(blobUrl, this.storageClient))
+                {
+                    throw new ArgumentOutOfRangeException("blobUrl", "Disk's page blob does not exist: '" + blobUrl + "'.");
+                }
+            }
+        }
+
+        private bool CheckIfPageBlobExists(string blobUrlInput, StorageManagementClient storageClient)
+        {
+            if (storageClient == null)
+            {
+                throw new ArgumentNullException("storageClient");
+            }
+
+            BlobUri blobUri = null;
+            bool result = BlobUri.TryParseUri(new Uri(blobUrlInput), out blobUri);
+            if (blobUri == null)
+            {
+                throw new ArgumentOutOfRangeException(blobUrlInput, "Input Blob Url is invalid: '" + blobUrlInput + "'.");
+            }
+
+            CloudPageBlob pageBlobRef = new CloudPageBlob(blobUri.Uri);
```

Comment:
Again, using the more gernic version with IStorageContext would mean less cusomt code needed here